solutions/projectstaffing/jgraph/core/src/main/scala/com/microsoft/graphdataconnect/skillsfinder/config/UserIdExtractionFilter.scala

```diff
@@ -12,26 +12,37 @@ import com.microsoft.graphdataconnect.skillsfinder.service.UserService
 import javax.servlet.http.{HttpFilter, HttpServletRequest, HttpServletResponse}
 import javax.servlet.{FilterChain, ServletException}
 import org.slf4j.{Logger, LoggerFactory}
-import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.{Autowired, Value}
 import org.springframework.context.annotation.Configuration
 import org.springframework.http.HttpStatus
 
 @Configuration
 class UserIdExtractionFilter(@Autowired val userService: UserService) extends HttpFilter {
   private val log: Logger = LoggerFactory.getLogger(classOf[UserIdExtractionFilter])
+  private val arrayAnonymousUserInfo = Map("accesToken" -> "", "idToken" -> "", "refreshToken" -> "", "userId" -> "test@anonymous.com")
+
+  @Value("${anonymous.authentication}")
+  var anonymousUser: Boolean  = _
 
   @throws[IOException]
   @throws[ServletException]
   override protected def doFilter(request: HttpServletRequest, response: HttpServletResponse, filterChain: FilterChain): Unit = {
     try {
-      val userId: Option[String] = getUserIdFromHttpServletRequest(request)
 
-      if (userId.isDefined) {
+      if (anonymousUser) {
```

Comment:
Maybe a cleaner solution would have been to create a new spring profile (let's say anonymous-auth) and define new implementations for MdcFilter, UserIdExtractionFilter and UserService that get instantiated only when that profile is active, in order to avoid all the ifs. But for now I think this will do.

Reply:
indeed, that would be a very elegant approach, however, this is low prio. So for now, the current approach is good enough.